From: Bjorn Helgaas <bhelgaas@google.com>
Date: Mon, 30 Nov 2015 19:35:53 -0600
Subject: PCI/MSI: Remove empty pci_msi_init_pci_dev()
References: bnc#993388,FATE#321732
Patch-Mainline: v4.5-rc1
Git-commit: 128fc68ca1c3c73ed35ef9c7c50b8972de40322f

4a7cc8316705 ("genirq/MSI: Move msi_list from struct pci_dev to struct
device") removed the contents of pci_msi_init_pci_dev().  All
implementation of it are now empty, so remove it completely.

Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/pci/msi.c   | 4 ----
 drivers/pci/pci.h   | 2 --
 drivers/pci/probe.c | 3 ---
 3 files changed, 9 deletions(-)

diff --git a/drivers/pci/msi.c b/drivers/pci/msi.c
index 62741e9..c41d56a 100644
--- a/drivers/pci/msi.c
+++ b/drivers/pci/msi.c
@@ -1029,10 +1029,6 @@ int pci_msi_enabled(void)
 }
 EXPORT_SYMBOL(pci_msi_enabled);
 
-void pci_msi_init_pci_dev(struct pci_dev *dev)
-{
-}
-
 /**
  * pci_enable_msi_range - configure device's MSI capability structure
  * @dev: device to configure
diff --git a/drivers/pci/pci.h b/drivers/pci/pci.h
index 82400b23..e8d0d07 100644
--- a/drivers/pci/pci.h
+++ b/drivers/pci/pci.h
@@ -145,10 +145,8 @@ extern unsigned int pci_pm_d3_delay;
 
 #ifdef CONFIG_PCI_MSI
 void pci_no_msi(void);
-void pci_msi_init_pci_dev(struct pci_dev *dev);
 #else
 static inline void pci_no_msi(void) { }
-static inline void pci_msi_init_pci_dev(struct pci_dev *dev) { }
 #endif
 
 static inline void pci_msi_set_enable(struct pci_dev *dev, int enable)
diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index 771b8bb..19764bb 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -1661,9 +1661,6 @@ static void pci_init_capabilities(struct pci_dev *dev)
 	/* Enhanced Allocation */
 	pci_ea_init(dev);
 
-	/* MSI/MSI-X list */
-	pci_msi_init_pci_dev(dev);
-
 	/* Setup MSI caps & disable MSI/MSI-X interrupts */
 	pci_msi_setup_pci_dev(dev);
 
-- 
1.8.5.6

